pkg/scheduler/server/server.go

```diff
@@ -273,3 +302,91 @@ func clientEndpoints(initialPeersListIP []string, idToPort map[string]string) []
 	}
 	return clientEndpoints
 }
+
+// runJobWatcher (dynamically) watches for (client) sidecar connections and adds them to the connection pool.
+func (s *Server) runJobWatcher(ctx context.Context) error {
+	log.Infof("Starting job watcher")
+
+	s.jobWatcherWG.Add(2)
+
+	// Goroutine for handling sidecar connections
+	go func() {
+		defer log.Info("Sidecar connections goroutine shutting down.")
+		defer s.jobWatcherWG.Done()
+		s.handleSidecarConnections(ctx)
+	}()
+
+	// Goroutine for handling job streaming at trigger time
+	go func() {
+		defer log.Info("Job streaming goroutine shutting down.")
```

Comment:
Please don't use "goroutine" or similar in log lines as this likely doesn't mean anything to the user, and is an implementation detail.

```suggestion
		defer log.Info("Job streaming shutting down.")
```